src/v2/view-builder/views/ov/NumberChallengePushView.js

```diff
@@ -0,0 +1,122 @@
+import { _, loc, View } from 'okta';
+import hbs from 'handlebars-inline-precompile';
+import BaseForm from '../../internals/BaseForm';
+import polling from '../shared/polling';
+import { WARNING_TIMEOUT } from '../../utils/Constants';
+
+let correctAnswer = '';
+const warningTemplate = View.extend({
+  className: 'okta-form-infobox-warning infobox infobox-warning',
+  template: hbs`
+    <span class="icon warning-16"></span>
+     <p>{{i18n
+      code="oie.numberchallenge.warning"
+      bundle="login"
+      $1="<a href='#' class='resend-number-challenge'>$1</a>"}}
+    </p>
+  `
+});
+
+const numberChallengeTemplate = View.extend({
+  className: 'number-challenge-section',
+  template: hbs`
+     <p>
+     {{i18n
+      code="oie.numberchallenge.subtitle"
+      bundle="login"
+      arguments="correctAnswer"
+      $1="<span class='strong'>$1</span>"}}
+     </p>
+    <div class="phone">
+      <div class="phone--body">
+        <div class="phone--screen">
+          <span class="phone--number" data-se="challenge-number">{{correctAnswer}}</span>
+        </div>
+        <div class="phone--home-button"></div>
+      </div>
+    </div>
+  `,
+  getTemplateData () {
+    return {
+      correctAnswer
+    };
+  },
+});
+
+
+const Body = BaseForm.extend(Object.assign(
+  {
+    noButtonBar: true,
+
+    className: 'okta-verify-number-challenge',
+
+    events: {
+      'click a.resend-number-challenge': 'handleResendNumberChallenge'
+    },
+
+    handleResendNumberChallenge () {
+      this.options.appState.trigger('invokeAction', 'currentAuthenticator-resend');
+      this.showWarningWithDelay();
+    },
+
+    title () {
+      return loc('oie.okta_verify.push.sent', 'login');
+    },
+
+    initialize () {
+      BaseForm.prototype.initialize.apply(this, arguments);
+      this.listenTo(this.model, 'error', this.stopPoll);
```

Comment:
Instead of adding a listener (which is at many places), override `TriggerAfterError()`

Reply:
Curious what is the advantage of using `TriggerAfterError`?

Reply:
we have the ability, so we can use it. We can do more in it apart from just `stopPoll`.. just like we override `remove` instead of adding a listener